app/models/annotation/nml/NmlWriter.scala

```diff
@@ -190,4 +198,38 @@ object NmlWriter {
       }
     }
   }
+
+  def writeMetaData(annotation: Annotation)(implicit writer: XMLStreamWriter) = {
+    val userNameFuture = annotation.user.map(_.name).getOrElse("No user")
+    var userName = ""
+    userNameFuture.onComplete {
+      case Success(name) => userName = name
+      case Failure(e) => userName = "No user"
+    }
+
+    Xml.withinElementSync("meta") {
+      writer.writeAttribute("name", "writer")
+      writer.writeAttribute("content", "NmlWriter.scala")
+    }
+    Xml.withinElementSync("meta") {
+      writer.writeAttribute("name", "writerGitCommit")
+      writer.writeAttribute("content", webknossos.BuildInfo.commitHash)
+    }
+    Xml.withinElementSync("meta") {
+      writer.writeAttribute("name", "timestamp")
+      writer.writeAttribute("content", DateTime.now().getMillis.toString)
+    }
+    Xml.withinElementSync("meta") {
+      writer.writeAttribute("name", "annotationId")
+      writer.writeAttribute("content", annotation.id)
+    }
+    Xml.withinElementSync("meta") {
+      writer.writeAttribute("name", "username")
+      writer.writeAttribute("content", userName)
+    }
+    Xml.withinElementSync("meta") {
+      writer.writeAttribute("name", "taskId")
+      writer.writeAttribute("content", annotation._task.get.stringify)
```

Comment:
I would suggest to not write the tags at all if the field is not present. Rather, annotation._task.map(taskId => Xml.withinElementSync...).
The .get here will possibly throw an exception for explorative annotations?